Feat/add more datasets

saattrupdan opened the pull request:
### Added
- Added the following new datasets:
    - `fone`, a Faroese NER dataset, which replaces the previous `wikiann-fo` dataset.
    - `dansk`, a Danish NER dataset, which complements the previous `dane` dataset.
    - `norquad`, a Norwegian question answering dataset, which replaces the previous `scandiqa-no` dataset.
    - `mmlu-da`, `mmlu-sv`, `mmlu-de`,  and `mmlu-nl`, being GPT-3.5-turbo translated versions of MMLU, the multiple choice dataset testing the knowledge of generative models.
    - `arc-da`, `arc-sv`, `arc-de`,  and `arc-nl`, being GPT-3.5-turbo translated versions of HellaSwag, the multiple choice dataset testing the knowledge of grade-school science questions.
    - `hellaswag-da`, `hellaswag-sv`, `hellaswag-de`,  and `hellaswag-nl`, being GPT-3.5-turbo translated versions of HellaSwag, the multiple choice dataset testing the common sense reasoning of generative models.

This closes #94, #93, #92, #91 and #89.

KennethEnevoldsen commented on src/scripts/create_dansk.py:
```diff
@@ -0,0 +1,176 @@
+    # This mapping is taken from Tedeschi, Simone, et al. "WikiNEuRal: Combined neural
+    # and knowledge-based silver data creation for multilingual NER." Findings of the
+    # Association for Computational Linguistics: EMNLP 2021. 2021.
+    ontonotes_tag_to_conll_tag_mapping: dict[str, str] = {
+        "CARDINAL": "O",
+        "DATE": "O",
+        "EVENT": "MISC",
+        "FACILITY": "LOC",
+        "GPE": "LOC",
+        "LANGUAGE": "MISC",
+        "LAW": "O",
+        "LOCATION": "LOC",
+        "MONEY": "O",
+        "NORP": "MISC",
+        "ORDINAL": "O",
+        "ORGANIZATION": "ORG",
+        "PERCENT": "O",
+        "PERSON": "PER",
+        "PRODUCT": "MISC",
+        "QUANTITY": "O",
+        "TIME": "O",
+        "WORK OF ART": "MISC",
+    }
```
Why is it that we map to the conll2003 format? It seems simplifying without adding any benefits (in fact, differing from the 'standard' might be good at measuring weaknesses with LLMs, which perform poorly on multilevel classification, especially if there are a lot of labels).

saattrupdan replied:
Hmm a really good point actually. Initially it was to be able to compare across languages, but since it seems like we're moving away from that anyway, I don't see why we can't change this too. This is also the case for several other NER datasets in that case (Swedish, Norwegian and Faroese all use different tagging systems).

KennethEnevoldsen replied:
Yea, I believe it is great to standardize e.g. when training a model but in our case keeping the original labels is probably better. Unsure if the prompts for NER are written specifically for conll2003?

saattrupdan replied:
Also, I think the NER module needs to change to reflect this change, somehow. Currently the label scheme is hardcoded, and the label scheme is also used in the few shot prompts. So not sure how to deal with it

KennethEnevoldsen replied:
I can't find a full overview of the prompt, but I would probably create the label section and then either hardcode it pr. dataset or add a description pr. label and then parse it in automatically. 

the label scheme should ideally not be hardcoded. Thought it might be ideal to commit this PR for now and just create an issue for this.

saattrupdan replied:
Added issue on it now: https://github.com/ScandEval/ScandEval/issues/96

KennethEnevoldsen commented on src/scripts/create_fone.py:
```diff
@@ -0,0 +1,114 @@
+    df.rename(columns={"ner_tags": "labels"}, inplace=True)
+
+    # Convert the NER tags from IDs to strings
+    ner_conversion_dict = {
+        0: "O",  # Original: B-Date
+        1: "B-LOC",  # Original: B-Location
+        2: "B-MISC",  # Original: B-Miscellaneous
+        3: "O",  # Original: B-Money
+        4: "B-ORG",  # Original: B-Organization
+        5: "O",  # Original: B-Percent
+        6: "B-PER",  # Original: B-Person
+        7: "O",  # Original: B-Time
+        8: "O",  # Original: I-Date
+        9: "I-LOC",  # Original: I-Location
+        10: "I-MISC",  # Original: I-Miscellaneous
+        11: "O",  # Original: I-Money
+        12: "I-ORG",  # Original: I-Organization
+        13: "O",  # Original: I-Percent
+        14: "I-PER",  # Original: I-Person
+        15: "O",  # Original: I-Time
+        16: "O",  # Original: O
+    }
```
same question as for dansk

saattrupdan replied:
Moving the discussion to that comment 🙂

saattrupdan commented on CHANGELOG.md:
```diff
@@ -23,13 +23,32 @@ and this project adheres to
+    - `mmlu-da`, `mmlu-sv`, `mmlu-de`,  and `mmlu-nl`, being GPT-3.5-turbo translated
+      versions of MMLU, the multiple choice dataset testing the knowledge of generative
+      models.
+    - `arc-da`, `arc-sv`, `arc-de`,  and `arc-nl`, being GPT-3.5-turbo translated
```
I completely agree regarding the bias. This was primarily to have _something_ to enable us to evaluate the models on these kinds of tasks, which should be replaced by gold standard datasets later.

And I didn't choose gpt-3.5, as we didn't do the translation - it was simply already freely available from the University of Oregon, so it was a start. 

We're currently looking into translation of the remaining Germanic languages (Norwegian, Icelandic and Faroese), as well as validation of the translations from native speakers. So see this as a start, and I think we should wait with a new ScandEval release until the translations are done and validated. 

Does that sound alright with you?

KennethEnevoldsen replied:
Sounds very reasonable. I would add that justification somewhere (changelog maybe?) such that it is made clear somewhere

saattrupdan replied:
Added it to the changelog now.

saattrupdan commented:
@KennethEnevoldsen Noticed there was an issue with the DANSK script, as it didn't allow for NER tags spanning multiple tokens - this has been fixed now.

AJDERS reviewed (approved):
LGTM!

KennethEnevoldsen commented:
> @KennethEnevoldsen Noticed there was an issue with the DANSK script, as it didn't allow for NER tags spanning multiple tokens - this has been fixed now.

The dataset or the code?

saattrupdan commented:
> > @KennethEnevoldsen Noticed there was an issue with the DANSK script, as it didn't allow for NER tags spanning multiple tokens - this has been fixed now.
> 
> The dataset or the code?

Both - I fixed both dataset and code. I removed all DANSK evaluations and started from scratch again after that fix.

KennethEnevoldsen commented:
Ahh so not an error in the original dataset
